features/device-config/ssh-scripting/src/main/java/org/opennms/features/deviceconfig/sshscripting/impl/SshScriptingServiceImpl.java

```diff
@@ -190,7 +198,22 @@ private SshInteractionImpl(
                 }
 
                 try {
-                    session.addPasswordIdentity(password);
+                    if (password != null) {
+                        session.addPasswordIdentity(password);
+                    }
+
+                    if (!Strings.isNullOrEmpty(authKey)) {
```

Comment:
If it is possible to add a test for this ?

Reply:
Tests added